packages/@react-stately/dnd/src/useDraggableCollectionState.ts

```diff
@@ -52,7 +52,7 @@ export function useDraggableCollectionState(props: DraggableCollectionOptions):
     // item is dragged. This matches native macOS behavior.
     let keys = new Set(
       selectionManager.isSelected(key)
-        ? new Set([...selectionManager.selectedKeys])
+        ? new Set([...selectionManager.selectedKeys].filter(key => !!collection.getItem(key)))
```

Comment:
Filters out any `selectedKeys` that don't exist in the collection when getting the dragged items. Didn't want to modify `selectionManger`'s `selectedKeys` getter since the user maybe tracking items that haven't been loaded into the collection (async loading/pagination case). However, we also don't want to perform a drag for items that don't exist in the collection IMO.